postgrenosql/src/test/java/com/yahoo/ycsb/postgrenosql/PostgreNoSQLDBClientTest.java

```diff
@@ -0,0 +1,305 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+package com.yahoo.ycsb.postgrenosql;
+
+import com.yahoo.ycsb.*;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.postgresql.Driver;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.Socket;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.*;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.not;
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assume.assumeNoException;
+
+/**
+ * PostgreNoSQL test client for YCSB framework.
+ */
+public class PostgreNoSQLDBClientTest {
+  private static final Logger LOG = LoggerFactory.getLogger(PostgreNoSQLDBClientTest.class);
+
+  /** The default port for PostgreSQL. */
+  private static final int DEFAULT_PORT = 5432;
+  private static final String DATABASE_NAME = "test";
+  private static final String DEFAULT_USER = "postgres";
+  private static final String DEFAULT_PWD = "postgres";
+
+  /** The properties settings */
+  private static final String HOST_NAME  = "localhost";
+  private static final String TEST_DB_URL = "jdbc:postgresql://" + HOST_NAME + ":" + DEFAULT_PORT + "/" + DATABASE_NAME;
+  private static final String TABLE_NAME = "usertable";
+  private static final int FIELD_LENGTH = 32;
+  private static final String FIELD_PREFIX = "FIELD";
+  private static final int NUM_FIELDS = 3;
+
+  private static Connection postgreSQLConnection = null;
+  private static PostgreNoSQLDBClient postgreNoSQLClient = null;
+
+  @BeforeClass
+  public static void setUp() {
+    // Check whether postgres is available
+    try (Socket socket = new Socket(HOST_NAME, DEFAULT_PORT)){
+      assertThat("Socket is not bound.", socket.getLocalPort(), not(-1));
+    } catch (IOException connectFailed) {
+      assumeNoException("PostgreSQL is not running. Skipping tests.", connectFailed);
+    }
+
+    Properties props = new Properties();
+    props.setProperty(PostgreNoSQLDBClient.CONNECTION_URL, TEST_DB_URL);
+    props.setProperty(PostgreNoSQLDBClient.CONNECTION_USER, DEFAULT_USER);
+    props.setProperty(PostgreNoSQLDBClient.CONNECTION_PASSWD, DEFAULT_PWD);
+    props.setProperty("user", DEFAULT_USER);
+    props.setProperty("password", DEFAULT_PWD);
+    props.setProperty(PostgreNoSQLDBClient.JDBC_AUTO_COMMIT, "true");
+
+    try{
+      postgreSQLConnection = new Driver().connect(TEST_DB_URL, props);
+      deleteTable();
```

Comment:
why have these extra create / delete table calls along with the create table call in the travis script? you should choose either doing the set up in travis (and add to the class javadoc that the test presumes you've set up a correct table) or do it here in the test. Personally, I think the "presume you have a table and use travis" is easier to maintain, but I don't feel strongly.